pkg/steps/new_e2e.go

```diff
@@ -66,14 +75,20 @@ func (s *newE2ETestStep) Run(ctx context.Context, dry bool) error {
 		return nil
 	}
 	var errs []error
-	if err := s.runSteps(ctx, s.pre); err != nil {
+	var err error
+	if err = s.runSteps(ctx, s.pre, ""); err != nil {
 		errs = append(errs, fmt.Errorf("%q pre steps failed: %v", s.name, err))
-	} else {
-		if err := s.runSteps(ctx, s.tests); err != nil {
+	}
+	var secret string
+	if secret, err = createSecret(s.secretClient, s.jobSpec.Namespace, s.artifactDir); err != nil {
+		errs = append(errs, fmt.Errorf("failed to create secret: %v", err))
+	}
+	if errs == nil {
+		if err = s.runSteps(ctx, s.tests, secret); err != nil {
 			errs = append(errs, fmt.Errorf("%q test steps failed: %v", s.name, err))
 		}
 	}
-	if err := s.runSteps(ctx, s.post); err != nil {
+	if err = s.runSteps(ctx, s.post, secret); err != nil {
```

Comment:
This should remain `err :=` ?

Reply:
Artifact from a previous implementation, all `err`s can be scoped now.